sound/soc/codecs/rt1011.c

```diff
@@ -1891,9 +1893,9 @@ static int rt1011_set_tdm_slot(struct snd_soc_dai *dai,
 	}
 
 	/* Tx slot configuration */
-	tx_slotnum = hweight_long(tx_mask);
-	first_bit = find_next_bit((unsigned long *)&tx_mask, 32, 0);
-	last_bit = find_last_bit((unsigned long *)&tx_mask, 32);
+	tx_slotnum = hweight_long(tx_mask_l);
+	first_bit = find_next_bit(&tx_mask_l, 32, 0);
+	last_bit = find_last_bit(&tx_mask_l, 32);
```

Comment:
`__fls()` but also see if any of the masks can be 0 and what you should do then.